addons/binding/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/handler/ExecHandler.java

```diff
@@ -53,191 +58,326 @@
-                        logger.debug("Exec [{}]: '{}'", "ERROR", line);
+                    try (InputStreamReader isr = new InputStreamReader(proc.getErrorStream());
+                            BufferedReader br = new BufferedReader(isr);) {
+                        String line = null;
+                        while ((line = br.readLine()) != null) {
+                            errorBuilder.append(line).append("\n");
+                            logger.debug("Exec [{}]: '{}'", "ERROR", line);
+                        }
+                        isr.close();
+                    } catch (IOException e) {
+                        logger.error("An exception occurred while reading the stderr when executing '{}' : '{}'",
+                                new Object[] { commandLine.toString(), e.getMessage() });
                     }
-                    isr.close();
-                } catch (IOException e) {
-                    logger.error("An exception occurred while reading the stderr when executing '{}' : '{}'",
-                            new Object[] { commandLine.toString(), e.getMessage() });
-                }
 
-                boolean exitVal = false;
-                try {
-                    exitVal = proc.waitFor(timeOut, TimeUnit.MILLISECONDS);
-                } catch (InterruptedException e) {
-                    logger.error("An exception occurred while waiting for the process ('{}') to finish : '{}'",
-                            new Object[] { commandLine.toString(), e.getMessage() });
-                }
+                    boolean exitVal = false;
+                    try {
+                        exitVal = proc.waitFor(timeOut, TimeUnit.MILLISECONDS);
+                    } catch (InterruptedException e) {
+                        logger.error("An exception occurred while waiting for the process ('{}') to finish : '{}'",
+                                new Object[] { commandLine.toString(), e.getMessage() });
+                    }
 
-                if (!exitVal) {
-                    logger.warn("Forcibly termininating the process ('{}') after a timeout of {} ms",
-                            new Object[] { commandLine.toString(), timeOut });
-                    proc.destroyForcibly();
-                }
+                    if (!exitVal) {
+                        logger.warn("Forcibly termininating the process ('{}') after a timeout of {} ms",
+                                new Object[] { commandLine.toString(), timeOut });
+                        proc.destroyForcibly();
+                    }
 
-                updateState(RUN, OnOffType.OFF);
-                updateState(EXIT, new DecimalType(proc.exitValue()));
+                    updateState(RUN, OnOffType.OFF);
+                    updateState(EXIT, new DecimalType(proc.exitValue()));
 
-                outputBuilder.append(errorBuilder.toString());
+                    outputBuilder.append(errorBuilder.toString());
```

Comment:
Maybe just add them to same `StringBuilder` right away.

Reply:
I prefer to keep them apart